lib/health_check.rb

```diff
@@ -0,0 +1,97 @@
+class HealthCheck
+  include ChefResource
+
+  OK = 'ok'.freeze
+  NOT_OK = 'not ok'.freeze
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  AUTHERROR = 'authentication error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = OK
+    @erchef = { status: REACHABLE }
+    @postgres = { status: REACHABLE }
+  end
+
+  #
+  # Do a general health check.
+  #
+  def check
+    erchef_health
+    erchef_authentication
+    postgres_health
+    overall
+  end
+
+  #
+  # Shortcut to see if all is well
+  #
+  def ok?
+    @status == OK
+  end
+
+  private
+
+  #
+  # Try to ping erchef as an indicator of general health
+  #
+  def erchef_health
+    erchef_health_metric do
+      chef.get_rest '_status'
+    end
+  end
+
+  #
+  # Try to authenticate with erchef to prove that oc-id is actually functioning
+  #
+  # TODO: in order for this to work, we need a test user, with
+  # username/password stored in the vault and accessible in Settings
+  #
+  def erchef_authentication
```

Comment:
My guess is that the complexity it adds outweighs the value it gives us right now, so let's delete it and ship for now and then add it at a time where it will be more valuable if that time ever comes.